src/analysis/analyze.rs

```diff
@@ -129,28 +129,14 @@ impl Header
         }
```

Comment:
It looks like you just cleaned up the if statements here, so I'm ok with these changes.

Reply:
some behavioral differences too, but certainly it's more correct now. if any slice was shorter than 7 bytes, it used to panic, but now it identifies those as "BIN" instead of ELF, PE, etc

Reply:
the fuzzer was getting tripped up on that, generating very little coverage, but lots of crashes